learn/guides/asyncapi-tool.md

```diff
@@ -16,15 +16,101 @@ The Ballerina AsyncAPI tool will make it easy for you to start the development o
+
+```yaml
+asyncapi: 2.1.0
+x-ballerina-event-identifier:
+ type: "body"
+ path: "event.type"
+components:
+ schemas:
+   GenericEventWrapper:
+     additionalProperties: true
+     description: Adapted from auto-generated content
+     properties:
+       event:
+         additionalProperties: true
+         properties:
+           event_ts:
+             title: When the event was dispatched
+             type: string
+           type:
+             title: The specific name of the event
+             type: string
+           text:
+             title: The message content
+             type: string
+         required:
+           - type
+           - event_ts
+         title: "The actual event, an object, which happened"
+         type: object
+     required:
+       - event
+     title: Standard event wrapper for the Events API
+     type: object
+channels:
+ app:
+   subscribe:
+     message:
+       oneOf:
+         - x-ballerina-event-type: "app_mention"
+           externalDocs:
+             description: Event documentation for `app_mention`
+           payload:
+             $ref: "#/components/schemas/GenericEventWrapper"
+           summary: Subscribe to only the message events that mention your app or bot
+           tags:
+             - name: allows_user_tokens
+             - name: app_event
+           x-scopes-required: []
+           x-tokens-allowed:
+             - user
+	   - x-ballerina-event-type: "app_rate_limited"
+           externalDocs:
+             description: Event documentation for `app_rate_limited`
+           payload:
+             $ref: "#/components/schemas/GenericEventWrapper"
+           summary: Indicates that your app's event subscriptions are being rate limited
+           tags:
+             - name: allows_user_tokens
+             - name: app_event
+             - name: allows_workspace_tokens
+           x-scopes-required: []
+           x-tokens-allowed:
+             - user
+             - workspace
+```
+
+There are custom tags in this YAML starting with `x-ballerina`. It is very important that these tags must be added to the AsyncAPI contract before using the tool. The usage of those tags will be as follows.
+
+1\. `x-ballerina-event-identifier` - When the listener receives an event from the event source (Slack is the event source in this scenario), there should be a way to identify the event type. This includes two parts, `type` and `path`.
+
+- type - Type can be either header or body. In other words, the type of event can be included in the payload either as an HTTP header or as an attribute in the body.
+
+  > **Note:** Currently, this tool supports only the body property. Hence, the path is equal to the JSON path of the attribute.
+
+- path - Path is equal to the header-name if the type is `header`, or the JSON path of the attribute if the type is `body`.
```

Comment:
```suggestion
- `path` - Path is equal to the header-name if the type is `header`, or the JSON path of the attribute if the type is `body`.
```

Reply:
Thanks a lot for the suggestion, Resolved